packages/cli/src/commands/push.ts

```diff
@@ -41,92 +46,123 @@ export async function handlePush(argv: {
   }
 
   const startedAt = performance.now();
-  const { entrypoint, destination, branchName, upsert } = argv;
+  const { destination, branchName, upsert } = argv;
 
-  if (!validateDestination(destination!)) {
+  if (
+    destination &&
+    !(validateDestination(destination) || validateDestinationWithoutOrganization(destination))
+  ) {
     exitWithError(
       `Destination argument value is not valid, please use the right format: ${yellow(
         '<@organization-id/api-name@api-version>',
       )}`,
     );
   }
 
-  const [organizationId, name, version] = getDestinationProps(destination!);
+  const [organizationId, name, version] = getDestinationProps(destination, config.organization);
 
-  try {
-    let rootFilePath = '';
-    const filePaths: string[] = [];
-    const filesToUpload = await collectFilesToUpload(entrypoint!);
-    const filesHash = hashFiles(filesToUpload.files);
-
-    process.stdout.write(
-      `Uploading ${filesToUpload.files.length} ${pluralize('file', filesToUpload.files.length)}:\n`,
+  if (!organizationId) {
+    return exitWithError(
+      `No organization provided, please use the right format: ${yellow(
+        '<@organization-id/api-name@api-version>',
+      )} or specify the 'organization' field in the config file.`,
     );
+  }
+  const entrypoint =
+    argv.entrypoint || (name && version && getApiEntrypoint({ name, version, config }));
+  if (name && version && !entrypoint) {
+    exitWithError(
+      `No entrypoint found that matches ${blue(
+        `${name}@${version}`,
+      )}. Please make sure you have provided the correct data in the config file.`,
+    );
+  }
 
-    let uploaded = 0;
-
-    for (let file of filesToUpload.files) {
-      const { signedUploadUrl, filePath } = await client.registryApi.prepareFileUpload({
-        organizationId,
-        name,
-        version,
-        filesHash,
-        filename: file.keyOnS3,
-        isUpsert: upsert,
-      });
-
-      if (file.filePath === filesToUpload.root) {
-        rootFilePath = filePath;
-      }
+  const apis = entrypoint ? { [`${name}@${version}`]: { root: entrypoint } } : config.apis;
```

Comment:
```suggestion
  // move here
  const apis = entrypoint ? { [`${name}@${version}`]: { root: getFallbackEntryPointsOrExit([entrypoint], config)[0] ) } } : config.apis;
```

Reply:
Checked. This works correctly without changes.